### DISCUSSION TITLE
Button to toggle Hide/Show Component

### DISCUSSION THREAD

**Discussion by Winniepoom** [Budibase/budibase#11465] [Help] [answered]
Hello I trying to make an Button to toggle show/hide my filter. as default I want it to be hide and when I click a button it show container which had a option picker for filtering 

![image](https://github.com/Budibase/budibase/assets/107398226/1c7745aa-1cf8-4a79-8978-56246acac922)
 
right now I can make it hide as default and can click to show filter but when I click filter button again it do nothing here my info.

![image](https://github.com/Budibase/budibase/assets/107398226/595172ba-8845-412c-b8ee-4e6b12c4be4a)

My button for toggle hide/show

![image](https://github.com/Budibase/budibase/assets/107398226/5bad0207-9f25-47d4-b6a9-1ee96a58dfc3)

My container

**Comment by ConorWebb96** [comment]
Hey @Winniepoom,

I'm not the best with handleBars so I will show you how to do this with JS.

1. Go to your screen and then click the on screen load button.
<img width="1464" alt="Screenshot 2023-08-08 at 08 15 42" src="https://github.com/Budibase/budibase/assets/126772285/a448a573-fa28-4ff0-8fd0-9af328d248eb">

2. Add your update state action to create the state we will use later.
3. Click the lighting bolt and switch to JS and `return false`
<img width="100%" alt="Screenshot 2023-08-08 at 08 16 36" src="https://github.com/Budibase/budibase/assets/126772285/3b4752be-2573-4545-9e2e-1e35c2e79c2b">

4. Click the container you wish to add this conditional UI to.
5. Click the tab on the far right of your screen `conditions`
6. Click the `configure conditions` button.
7. Configure your condition similar to what i've shown below;
<img width="100%" alt="Screenshot 2023-08-08 at 08 18 48" src="https://github.com/Budibase/budibase/assets/126772285/e6fcc0d7-653b-4cf3-b092-88ea86571449">

8. Configure the first setting on the above so its referencing your state value we created earlier.
9. Next click on the button you wish to toggle this UI on.
10. Click the `define actions` button
<img width="100%" alt="Screenshot 2023-08-08 at 08 24 16" src="https://github.com/Budibase/budibase/assets/126772285/00a3f7af-e1ad-4539-a9b0-596410b9725e">

11. Select the state we wish to update. Then paste the below code inside the `value` area of this `update state` action it after switching from handleBars to JS.
```
function oppositeBoolean(state) {
  return !state;
}

const state = $("State.filtersBoolean");
const oppositeState = oppositeBoolean(state);

return oppositeState;
```
12.  Save and this should do what you want.
13. A little bit extra you can amend the text to change its name from `show filters` to `hide filters` depending on the value of state.

<img width="1092" alt="Screenshot 2023-08-08 at 08 27 57" src="https://github.com/Budibase/budibase/assets/126772285/ce99e99a-8794-4438-957b-e712f96f165c">


I hope this helps.

**Comment by Winniepoom** [comment]
@ConorWebb96 Thank you for your answer but I following your demo it not seem to work correctly to me even I try to recreate a new project for testing , 

- as default it hide container - Work
- click button to show container - Work
- click button again to hide container - Not Work

My info 
Screen Config.
![image](https://github.com/Budibase/budibase/assets/107398226/0a7674a8-91e8-400e-ae8f-4ffc37ca1e7d)
![image](https://github.com/Budibase/budibase/assets/107398226/14740bd6-0fc4-4412-ba43-247a7fd3aaec)


Container Config 
![image](https://github.com/Budibase/budibase/assets/107398226/ba5842be-8f5c-4e2c-aa69-9dd0c58a98bb)

Button
![image](https://github.com/Budibase/budibase/assets/107398226/19f094a8-b8ab-4338-a610-ab03bc63b8b8)
![image](https://github.com/Budibase/budibase/assets/107398226/92a13965-598d-4e05-8be6-fcc44e898b86)

**Reply by ConorWebb96** [reply, marked as answer]
Hey @Winniepoom,

I've create dummy app here were it is working. Can you import this and let me know if it works for you.

[conditional ui-export-1691486024758.tar.gz](https://github.com/Budibase/budibase/files/12289516/conditional.ui-export-1691486024758.tar.gz)

I hope this helps.

**Comment by Winniepoom** [comment]
@ConorWebb96 It working now the reason why it now working it come from Screen define actions in key instead of "filtersBoolean" I'm using "filterBoolean" without "s" haha what a mistake Thank you so much for helping me.
